src/js/SelectFields/SelectField.js

```diff
@@ -1,307 +1,443 @@
+     * An optional style to apply to the select field's container (the menu).
      */
     style: PropTypes.object,
 
     /**
-     * An optional style to apply to the select field's container. This is the `Menu` component.
+     * An optional className to apply to the select field's container (the menu).
      */
     className: PropTypes.string,
 
     /**
-     * An optional style to apply to the select field's list of items that appear when opened.
+     * An optional style to apply to the menu's list.
      */
     listStyle: PropTypes.object,
 
     /**
-     * An optional className to apply to the select field's list of items that appear when opened.
+     * An optional className to apply to the menu's list.
      */
     listClassName: PropTypes.string,
 
     /**
-     * An optional style to apply to the select field itself.
+     * An optional style to apply to the select field's toggle.
+     */
+    toggleStyle: PropTypes.object,
+
+    /**
+     * An optional className to apply to the select field's toggle.
+     */
+    toggleClassName: PropTypes.string,
+
+    /**
+     * An optional style to apply to the `AccessibleFakeInkedButton` that is the trigger
+     * for the select field.
      */
     inputStyle: PropTypes.object,
 
     /**
-     * An optional className to apply to the select field itself.
+     * An optional className to apply to the `AccessibleFakeInkedButton` that is the trigger
+     * for the select field.
      */
     inputClassName: PropTypes.string,
 
     /**
-     * An optional value for the select field. This will require the `onChange` prop
-     * to be defined since it will be a controlled component.
+     * Boolean if the select field should be have the menu's list visible by default.
      */
-    value: controlled(PropTypes.oneOfType([
-      PropTypes.number,
-      PropTypes.string,
-    ]), 'onChange'),
+    defaultVisible: PropTypes.bool.isRequired,
 
     /**
-     * The default value for an uncontrolled select field.
+     * Boolean if the select field should have the menu's list visible. This will make
+     * the select field controlled and require the `onVisibilityChange` prop to be defined,
      */
-    defaultValue: PropTypes.oneOfType([
+    visible: controlled(PropTypes.bool, 'onVisibilityChange', 'defaultVisible'),
+
+    /**
+     * An optional function to call when the select field's menu has it's visibility changed. The callback
+     * will include the next visible state and the event that triggered it.
+     */
+    onVisibilityChange: PropTypes.func,
+
+    /**
+     * A list of `number`, `string`, or `object` that should be used to create `ListItem`
+     * in the menu's list. When it is an `object`, it will use the `dataLabel` prop as the
+     * `primaryText` and use the value of `dataValue`.
+     *
+     * @see {@link #dataLabel}
+     * @see {@link #dataValue}
+     */
+    menuItems: PropTypes.arrayOf(PropTypes.oneOfType([
```

Comment:
Should either rename this to `items` or rename `items` on `MenuButton` to `menuItems`